src/nvme/ioctl.h

```diff
@@ -384,22 +386,22 @@ int nvme_get_nsid(int fd, __u32 *nsid);
  * @uuidx:	UUID Index if controller supports this id selection method
  * @csi:	Command Set Identifier
  * @data:	User space destination address to transfer the data
- * @timeout:	Timeout in ms (0 for default timeout)
- * @result:	The command completion result from CQE dword0
  */
 struct nvme_identify_args {
 	int args_size;
 	int fd;
+	__u32 timeout;
+	__u32 rsvd12;
+	__u32 *result;
 	enum nvme_identify_cns cns;
 	__u32 nsid;
 	__u16 cntid;
 	__u16 nvmsetid;
 	__u16 domid;
 	__u8 uuidx;
+	__u8 rsvd40[5];
```

Comment:
What's the point of this 'rsvd49' structure?
We really should never do a simple 'memcpy' between this structure and the nvme identify command args, so no point in adding a fudge here.
If you need a reserved field for alignment a simple u8 variable should suffice, no?

Reply:
One thing we can do after introducing the reserved fields is use the pragma attribute packed which makes sure the compiler generates always the same layout.